### PR TITLE
Add smoke tests to make file and Pipeline configure

### PR DESCRIPTION
**Pull request by alexhung**
Considered a few ways to do this. Tried build tags but 1) it somehow doesn't work with TF tests, 2) would involve changing a lot of files, and 3) we may miss compilation issues with build tags.

Other ways like env vars or using `Testing.IsShort()` also involves updating lots of code, unnecessarily in my opinion.

So settled on a maintaining a list of test names in the make file for now.

### CODE REVIEW

File: pkg/artifactory/datasource/user/datasource_artifactory_user_test.go [hunk 1]
```diff
@@ -7,18 +7,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/acctest"
-	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/user"
 	"github.com/jfrog/terraform-provider-shared/test"
 	"github.com/jfrog/terraform-provider-shared/util"
 )
 
-func deleteUser(t *testing.T, name string) error {
```
Comment (danielmkn) [inline pkg/artifactory/datasource/user/datasource_artifactory_user_test.go:15]: Why is this deleted?

Comment (alexhung) [reply pkg/artifactory/datasource/user/datasource_artifactory_user_test.go:15]: Not used at all. This private func has been replaced by a public one https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/acctest/test.go#L282

**Review by shrajfr12 — APPROVED** [review]

**Review by danielmkn — DISMISSED** [review]

**Comment by alexhung** [comment]
Fixed the pipeline configuration after debugging with @danielmkn. Please review again, @danielmkn and @shrajfr12 Thx

**Review by danielmkn — APPROVED** [review]
